### DISCUSSION TITLE
Immich on Raspberry Pi 4 (4 GB RAM) causes whole server to crash.

### DISCUSSION THREAD

**Discussion by SayantanRC** [immich-app/immich#11644] [General]
I have been running Immich for 2 months. After a few recent updates (especially after v1.111), Immich is causing major problems.

I have external library to store all my photos and videos. Immich scans the library every night at 12 AM. And it puts so much load (load average exceeding 60) that my entire server crashes. I can't even SSH into it. I have to manually turn off power to the Raspberry Pi and turn it on again. This has been happening for a few nights now. If I keep the periodic scanning off, then everything is normal. But if I try to scan manually again, the same problem arises.

From `top`, I can see `ffmpeg` taking a fair amount of CPU usage. Although I have capped the containers to `cpus: 0.3`, even then my server crashes.

I have also set all job concurrencies to `1`.

Also my Raspberry Pi fan has recently broken down, so that might be exacerbating the situation.

Other than replacing the fan (which will take a few months), is there anything else I can do?

**Comment by mmomjian** [comment]
Immich doesn’t really have the option to crash your system, rather the underlying hardware problems are being exposed under heavy load.

**Comment by bo0tzz** [comment]
> Also my Raspberry Pi fan has recently broken down, so that might be exacerbating the situation.

If your system isn't adequately cooled, you've already lost ;)

**Comment by samolego** [comment]
I'm running this just fine on Pi 3b with 1 GB of RAM, though it's a bit slow on new jobs. But it doesn't crash the system in any way.

**Reply by SayantanRC** [reply]
How do you even achieve that? Do you keep machine learning off? Also what are your video transcoding settings?

**Reply by samolego** [reply]
I haven't done any special things ... Yup, I don't have ML enabled, haven't deepdived into it yet.

After setup it took about a day to process all the images (was unresponsive a lot at the time), but now it works (still not the fastest, but usable)

**Reply by SayantanRC** [reply]
For me it took over 3 days to go through 17000 photos and 300 videos. I had to keep all the job concurrencies reduced to `1`, otherwise my server would just freeze. This is when my Raspberry Pi fan was perfectly working.

But yes I have ML on.

**Reply by tdigu** [reply]
I am using a rpi4 2gb ram myself. I have a m.2 to USB as my os drive and an external 3.5 4tb to USB for media. I have everything set for 1 concurrent. It takes days to initially process everything. I forgot which case I have but it's an all aluminum case that acts as a heatsink and it's got 2 small fans. So far it's been doing great even when it was processing all my pictures and movies temps were about 110f. It wasn't the fastest thing but I also run homeassistant and both were running. It's a real value setup the most expensive component was the 4tb hard drive.

**Reply by harmpter** [reply]
@tdigu I'm trying to copy your setup, can I ask which Add-on you used to run Immich in Home Assistant? Assuming you even run HAOS - if you do not then my question is mute. I have a RPi 4 2GB ram, run HAOS off of a 1TB SSD, which I also want to use as photo storage in the /media directory. I just recently heard about Immich and am now excited to read that you made it work on 2GB ram. I will use my old gaming computer to support with heavy initial computation.

**Comment by tdigu** [comment]
i am running it in a docker container, along with home assistant. My base
OS is raspbian 64 bit. My RPI is a  rpi4 with 4 gigs of ram, I dont know if
it will run with only 2 gigs. I did have to trim down a lot of the settings
for congruity setting almost everything to 1.

**Reply by harmpter** [reply]
Ah, okay that makes sense, thank you. 🙏 I will try and make it work on 2gb, if I find an add-on. I think setting all concurrencies to 1 is not an issue for me as it only makes the jobs slower as far as I understand. I read people noticed immich only using about 400mb of RAM and I usually have 1gb sitting around, so I could get away with it, maybe.

**Reply by Prolonz** [reply]
I've run immich on a pi 4 with 2 GB ram. It was crashing whenever I uploaded new photos even with all jobs on 1. After I implemented the global job limit originally posted here [[discussions/3625]](https://github.com/immich-app/immich/discussions/3625#discussioncomment-14815768) it runs well. I hope a native solution is implemented some day.

**Reply by harmpter** [reply]
thanks, you probably saved me hours and hours of headaches! 🙏 Are you on HAOS?

**Reply by Prolonz** [reply]
No problem!

I do run HA but in docker on a Pi 4 with 8GB. I recently transferred immich to the Pi 4 with 8GB and it runs fine. I run those two and several other containers all using about 2.5GB ram. On 2 GB you might be able to get away with both.
